src/operator/operator.go

```diff
@@ -0,0 +1,746 @@
+		return ltNumeric(a, b), nil
+	}
+	if bothString(a, b) {
+		return a.(string) <= b.(string), nil
+	}
+
+	return nil, fmt.Errorf("given elements are not floor ltable")
+}
+
+func Gt(a any, b any) (any, error) {
+	if i, ok := a.(IGt); ok {
+		return i.Gt(b)
+	}
+	if i, ok := b.(ILt); ok {
+		return i.Lt(a)
+	}
+	if IsNumeric(a) && IsNumeric(b) {
+		return gtNumeric(a, b), nil
+	}
+	if bothString(a, b) {
+		return a.(string) >= b.(string), nil
+	}
+
+	return nil, fmt.Errorf("given elements are not floor gtable")
+}
+
+func Bool(a any) (bool, error) {
+	if i, ok := a.(IBool); ok {
+		return i.Bool()
+	}
+	return !reflect.ValueOf(a).IsZero(), nil
+}
+
+func Not(a any) (bool, error) {
+	b, err := Bool(a)
+	if err != nil {
+		return b, err
+	}
+	return !b, nil
+}
+
+func Pos(a any) (any, error) {
+	if i, ok := a.(IPos); ok {
+		return i.Pos()
+	}
+	if IsNumeric(a) {
+		return a, nil
+	}
+	return nil, fmt.Errorf("given element is not posable")
+}
+
+func Neg(a any) (any, error) {
+	if i, ok := a.(INeg); ok {
+		return i.Neg()
+	}
+	if IsNumeric(a) {
+		return multiplyNumeric(a, -1), nil
+	}
+	return nil, fmt.Errorf("given element is not negable")
+}
+
+func Contains(a, b any) (bool, error) {
+	if i, ok := a.(IContains); ok {
+		return i.Contains(b)
+	}
+	if bothString(a, b) {
+		return strings.Contains(a.(string), b.(string)), nil
+	}
+	switch reflect.TypeOf(a).Kind() {
+	case reflect.Array, reflect.Slice:
+		aV := reflect.ValueOf(a)
+		for i := 0; i < aV.Len(); i++ {
+			r, err := Eq(aV.Index(i).Interface(), b)
+			if err != nil {
+				return false, err
+			}
+			if r == true {
+				return true, nil
+			}
+		}
+		return false, nil
+	case reflect.Map:
+		aV := reflect.ValueOf(a)
+		iter := aV.MapRange()
+		for iter.Next() {
+			r, err := Eq(iter.Key().Interface(), b)
+			if err != nil {
+				return false, err
+			}
+			if r == true {
+				return true, nil
+			}
+		}
+		return false, nil
+	}
+	return false, fmt.Errorf("elements are not cointainable ")
+}
+
+func bothString(a, b any) bool {
+	_, aOk := a.(string)
+	_, bOk := b.(string)
+	return aOk && bOk
+}
+
+func addSlices(a, b any) []interface{} {
+	aV := reflect.ValueOf(a)
+	bV := reflect.ValueOf(b)
+
+	ret := make([]interface{}, 0, aV.Len()+bV.Len())
+	for i := 0; i < aV.Len(); i++ {
+		ret = append(ret, aV.Index(i).Interface())
+	}
+	for i := 0; i < bV.Len(); i++ {
+		ret = append(ret, bV.Index(i).Interface())
+	}
+	return ret
+}
+
+func mulStrByInt(a string, b int64) string {
+	res := ""
+	for i := int64(0); i < b; i++ {
+		res += a
+	}
+	return res
+}
+
+func mulSliceByInt(a any, b int64) interface{} {
+	t, v := reflect.TypeOf(a), reflect.ValueOf(a)
+	c := reflect.MakeSlice(t, v.Len()*int(b), v.Len()*int(b))
+	for i := 0; i < int(b); i++ {
+		reflect.Copy(c.Slice(i*v.Len(), (i+1)*v.Len()), v)
+	}
+	return c.Interface()
+}
+
+func multiplyNumeric(a any, b any) any {
+	res, _ := opNumeric(a, b, func(a any, b any) (any, error) {
+		switch v := a.(type) {
+		case int64:
+			return v * b.(int64), nil
```

Comment:
what if b is in fact uint64? won't the type assert fail here?

Reply:
THe way opNumeric is structured guarantees that a b are of the same type